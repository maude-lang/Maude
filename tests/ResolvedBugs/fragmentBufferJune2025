#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < $srcdir/fragmentBufferJune2025.maude -no-banner -no-advise \
  > fragmentBufferJune2025.out 2>&1

diff $srcdir/fragmentBufferJune2025.expected fragmentBufferJune2025.out > /dev/null 2>&1
