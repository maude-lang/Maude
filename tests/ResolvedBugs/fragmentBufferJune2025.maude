***
***	We used to clear the sort fragment buffer after we finished recognizing
***	a sort, but in the case of an error it would be left uncleared and cause
***	spurious errors.
***

fmod FOO is
*** ] is not allowed in sort names and causes the parsing of the structured
*** sort to fail with a bison syntax error.
  sort s{]} .
endfm

fmod BAR is
*** The tokens s { stashed in the uncleared fragment buffer ended up prepended
*** on to Bar.
   sort Bar .
   op c : -> Bar .
endfm

show sorts .
