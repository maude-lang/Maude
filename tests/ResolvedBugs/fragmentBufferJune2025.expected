Warning: <standard input>, line 10 (fmod FOO): syntax error
sort Bool .
sort Bar .
Bye.
